docker/tarfile/dest.go

```diff
@@ -34,11 +34,15 @@ type Destination struct {
 
 // NewDestination returns a tarfile.Destination for the specified io.Writer.
 func NewDestination(dest io.Writer, ref reference.NamedTagged) *Destination {
+	repoTags := []reference.NamedTagged{}
+	if ref != nil {
+		repoTags = append(repoTags, ref)
+	}
 	return &Destination{
 		writer:    dest,
 		tar:       tar.NewWriter(dest),
 		reference: ref,
```

Comment:
Please drop the `reference` field, it is no longer used by anything.

Reply:
fixed.